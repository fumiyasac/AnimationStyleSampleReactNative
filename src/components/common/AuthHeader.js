/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react';

import {
  StyleSheet
} from 'react-native';

import {
  Header,
  Body,
  Title
} from 'native-base';

export default class AuthHeader extends React.Component {

  // MARK: - Rendering Components

  render() {
    const { title } = this.props;
    return (
      <Header iosBarStyle="light-content" style={styles.headerBackgroundColor} hasTabs>
        <Body>
          <Title style={styles.titleColor}>{title}</Title>
        </Body>
      </Header>
    );
  };
}

// MARK: - Component Styles

const styles = StyleSheet.create({
  headerBackgroundColor: {
    backgroundColor: '#4169e1'
  },
  titleColor: {
    color: '#ffffff'
  }
});