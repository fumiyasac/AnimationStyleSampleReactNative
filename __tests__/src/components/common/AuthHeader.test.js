import 'react-native';
import React from 'react';
import AuthHeader from '../../../../src/components/common/AuthHeader';

import renderer from 'react-test-renderer';

it('AuthHeaderコンポーネントが正しく表示されること', () => {

  let stubObject = { title: "登録・認証" };

  const tree = renderer.create(
    <AuthHeader title={stubObject.title} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
  