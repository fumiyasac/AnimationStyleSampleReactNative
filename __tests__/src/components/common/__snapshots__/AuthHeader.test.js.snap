// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHeaderコンポーネントが正しく表示されること 1`] = `
<View>
  <View
    hasTabs={true}
    iosBarStyle="light-content"
    style={
      Array [
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": null,
          "elevation": 0,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 15,
          "right": 0,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "top": 0,
        },
        Object {
          "backgroundColor": "#4169e1",
        },
      ]
    }
    virtual={undefined}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
        }
      }
      virtual={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "600",
              "textAlign": "center",
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
        virtual={undefined}
      >
        登録・認証
      </Text>
    </View>
  </View>
</View>
`;
